Factorize a bit the code that change the zone on SK2 + UP/DOWN.

f1rmb opened the pull request.

rogerclarkmelbourne commented:
The code that does this

		if (menuDisplayQSODataState != QSO_DISPLAY_DEFAULT_SCREEN)
		{
			menuDisplayQSODataState = QSO_DISPLAY_DEFAULT_SCREEN;
		}


Seems strange.

You may as well always set 

menuDisplayQSODataState = QSO_DISPLAY_DEFAULT_SCREEN;

As its probably faster, as the compiler already has to fetch all the data and do the comparison etc.

rogerclarkmelbourne commented:
BTW. What problem does this fix  ?

f1rmb commented:
Hi Roger,

This fixes the blank screen if a QSO is running while switching to another zone (just like the bug we had using SK2 + UP/DOWN, forgot about the Zone menu.
Anyway, I've updated the PR, now menuDisplayQSODataState is set to force to redraw the display, regardless of its value (also changed the if (..) from the previous PR).

Cheers.
...
Daniel

rogerclarkmelbourne commented:
OK. Thanks

I"ll test it this evening
